internal/stdlib/syscall/exec_windows.go

```diff
@@ -299,68 +303,67 @@ func StartProcess(argv0 string, argv []string, attr *ProcAttr) (pid int, handle
 
 	var argvp *uint16
 	if len(cmdline) != 0 {
-		argvp, err = UTF16PtrFromString(cmdline)
+		argvp, err = windows.UTF16PtrFromString(cmdline)
 		if err != nil {
 			return 0, 0, err
 		}
 	}
 
 	var dirp *uint16
 	if len(attr.Dir) != 0 {
-		dirp, err = UTF16PtrFromString(attr.Dir)
+		dirp, err = windows.UTF16PtrFromString(attr.Dir)
 		if err != nil {
 			return 0, 0, err
 		}
 	}
 
-	var maj, min, build uint32
-	rtlGetNtVersionNumbers(&maj, &min, &build)
+	maj, min, _ := windows.RtlGetNtVersionNumbers()
 	isWin7 := maj < 6 || (maj == 6 && min <= 1)
 	// NT kernel handles are divisible by 4, with the bottom 3 bits left as
 	// a tag. The fully set tag correlates with the types of handles we're
 	// concerned about here.  Except, the kernel will interpret some
 	// special handle values, like -1, -2, and so forth, so kernelbase.dll
 	// checks to see that those bottom three bits are checked, but that top
 	// bit is not checked.
-	isLegacyWin7ConsoleHandle := func(handle Handle) bool { return isWin7 && handle&0x10000003 == 3 }
+	isLegacyWin7ConsoleHandle := func(handle windows.Handle) bool { return isWin7 && handle&0x10000003 == 3 }
 
-	p, _ := GetCurrentProcess()
+	p := windows.CurrentProcess()
 	parentProcess := p
 	if sys.ParentProcess != 0 {
 		parentProcess = sys.ParentProcess
 	}
-	fd := make([]Handle, len(attr.Files))
+	fd := make([]windows.Handle, len(attr.Files))
 	for i := range attr.Files {
 		if attr.Files[i] > 0 {
 			destinationProcessHandle := parentProcess
 
 			// On Windows 7, console handles aren't real handles, and can only be duplicated
 			// into the current process, not a parent one, which amounts to the same thing.
-			if parentProcess != p && isLegacyWin7ConsoleHandle(Handle(attr.Files[i])) {
+			if parentProcess != p && isLegacyWin7ConsoleHandle(windows.Handle(attr.Files[i])) {
 				destinationProcessHandle = p
 			}
 
-			err := DuplicateHandle(p, Handle(attr.Files[i]), destinationProcessHandle, &fd[i], 0, true, DUPLICATE_SAME_ACCESS)
+			err := windows.DuplicateHandle(p, windows.Handle(attr.Files[i]), destinationProcessHandle, &fd[i], 0, true, windows.DUPLICATE_SAME_ACCESS)
 			if err != nil {
 				return 0, 0, err
 			}
-			defer DuplicateHandle(parentProcess, fd[i], 0, nil, 0, false, DUPLICATE_CLOSE_SOURCE)
+			defer windows.DuplicateHandle(parentProcess, fd[i], 0, nil, 0, false, windows.DUPLICATE_CLOSE_SOURCE) // nolint: errcheck
 		}
 	}
-	si := new(_STARTUPINFOEXW)
-	si.ProcThreadAttributeList, err = newProcThreadAttributeList(2)
+	si := new(windows.StartupInfoEx)
+	si.ProcThreadAttributeList, err = windows.NewProcThreadAttributeList(2)
 	if err != nil {
 		return 0, 0, err
 	}
-	defer deleteProcThreadAttributeList(si.ProcThreadAttributeList)
+	defer si.ProcThreadAttributeList.Delete()
```

Comment:
is it ok, that `Delete()` is doing a bit more than the old `deleteProcThreadAttributeList` call?

Reply:
Yea, seems fine to me. I'm wondering why they didn't go with the x/sys/windows impl (not actually using the package, but just the impl) for this.